StardewArchipelago/Locations/CodeInjections/Modded/SVE/SVEShopStockModifier.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using StardewArchipelago.Archipelago;
+using StardewArchipelago.Constants.Modded;
+using StardewArchipelago.Constants.Vanilla;
+using StardewArchipelago.Stardew;
+using StardewModdingAPI;
+using StardewModdingAPI.Events;
+using StardewValley;
+using StardewValley.GameData.Shops;
+
+namespace StardewArchipelago.Locations.ShopStockModifiers
+{
+    public class SVEShopStockModifier : ShopStockModifier
```

Comment:
Since this class isn't used yet, I assume you're not done with it, but here's a couple pointers just in case
- You could replace all 3 weapons in one method call if you just defined a tiny list of "shops to check" and another list of "weapons to replace".
- As mentioned earlier, the QualifiedItemId part is wrong. Or, more specifically, it might work, but that's because ShopItem.Id is **not** necessarily an item id. It is an identifier within that shop. It is **fine** to find the items you need to replace using that ID, but they shouldn't be called Qualified Item Ids and shouldn't be in that class if that's the case, to avoid confusion. ShopItems also have a field called "ItemId" which might actually hold a qualified item id, so maybe you want to use this instead. Your call.

Reply:
Combining all three I'm not 100% on in terms of understanding what you mean.  But you can look it over and tell me if its fine enough.

Reply:
The comment about all 3 weapons in one call definitely isn't done.
Here's what I have in mind:
```
        private static readonly string[] _shopsWithTemperedWeapons = new []{"FlashShifter.StardewValleyExpandedCP_AlesiaVendor", "FlashShifter.StardewValleyExpandedCP_IsaacVendor"};
        
        // [...]
        
        public override void OnShopStockRequested(object sender, AssetRequestedEventArgs e)
        {
            if (!AssetIsShops(e))
            {
                return;
            }

            e.Edit(asset =>
            {
                var shopsData = asset.AsDictionary<string, ShopData>().Data;
                ReplaceTemperedWeapons(shopsData);

                var bearShop = shopsData["FlashShifter.StardewValleyExpandedCP_BearVendor"];
                MakeBearBarter(bearShop);
            },
                AssetEditPriority.Late
            );
        }

        private void ReplaceTemperedWeapons(IDictionary<string, ShopData> shops)
        {
            const string daggerLocationName = "Tempered Galaxy Dagger";
            const string swordLocationName = "Tempered Galaxy Sword";
            const string hammerLocationName = "Tempered Galaxy Hammer";
            var weaponIdsToLocations = new Dictionary<string, string>
            {
                { ModItemIds.TEMPERED_DAGGER, daggerLocationName},
                { ModItemIds.TEMPERED_SWORD, swordLocationName},
                { ModItemIds.TEMPERED_HAMMER, hammerLocationName},
            };
            foreach (var shopName in _shopsWithTemperedWeapons)
            {
                var shopData = shops[shopName];
                for (var i = shopData.Items.Count - 1; i >= 0; i--)
                {
                    var item = shopData.Items[i];
                    if (weaponIdsToLocations.ContainsKey(item.Id))
                    {
                        var apShopItem = CreateArchipelagoLocation(item, weaponIdsToLocations[item.Id]);
                        shopData.Items.RemoveAt(i);
                        shopData.Items.Insert(i, apShopItem);
                    }
                }
            }
        }
```